### ISSUE TITLE
Problems with MoveToTrashButton

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/scenery-phet#746]
While working on TrashButton for https://github.com/phetsims/scenery-phet/issues/745, I discovered MoveToTrashButton.  It has several problems, listed below.  Assigning to @samreid, listed as the first author.

- [x] The default size is unusually small.  Compare its default size to other buttons shown in scenery-phet demo below.  This default size results in callers having to scale the button.  Which results in the next problem...

- [x] MoveToTrashButton provides no way to scale the icon. Usage sites are scaling the button (not the icon), which will also scale the lineWidth of the button stroke.  That results in either an inferior look, or a need to fiddle with lineWidth to try to make it look reasonable for a specific scale (a brittle practice).

- [x] MoveToTrashButton changes the defaults for options baseColor, cornerRadius, xMargin, and yMargin. Why? I'm guessing because it was migrated from a sim, was not properly generalized, and is still using options that were appropriate for that sim. Other buttons typically use the defaults -- as they should. Using different defaults makes it difficult to have a uniform look.  If I use MoveToTrashButton and want to match a tyical RectangularPushButton, then I have to copy the defaults from RectangularPushButton, and pass them to MoveToTrashButton -- not a good practice.   If the defaults for RectangularPushButton are not appropriate (baseColor comes to mind...), that's a different issue -- let's change the defaults, not override them here and there.

- [x] Why is `buttonAppearanceStrategy: ButtonNode.FlatAppearanceStrategy`?  The default is typically 3D, and a few sims change that on a limited basis.  Again, if I want to use this in a typical sim, I have to change it back to the RectangularPushButton default.

<img width="960" alt="screenshot_1704" src="https://user-images.githubusercontent.com/3046552/174690013-86b78b7d-730f-41f4-a8ff-60216b30b089.png">

**Comment by samreid** [comment]
> This default size results in callers having to scale the button.

I observed 4 call sites in sim code, using these options:

energy-skate-park/EnergyBarGraph:

```ts
    const clearThermalButton = new MoveToTrashButton( {
      arrowColor: EnergySkateParkColorScheme.thermalEnergy,
      tandem: tandem.createTandem( 'clearThermalButton' ),
      listener: skater.clearThermal.bind( skater ),
      scale: 0.86
    } );
```

energy-skate-park/PieChartLegend:

```ts
    const clearThermalButton = new MoveToTrashButton( {
      arrowColor: EnergySkateParkColorScheme.thermalEnergy,
      tandem: tandem.createTandem( 'clearThermalButton' ),
      listener: clearThermal,
      centerX: thermalLabel.centerX,
      y: thermalLabel.bottom + 15,
      scale: 0.8
    } );
```

masses-and-springs/EnergyGraphAccordionBox:

```ts
    const clearThermalButton = new MoveToTrashButton( {
      arrowColor: ORANGE_COLOR,
      listener: () => {

        // We are setting a new initial total energy here because the thermal energy bar acts as if the system has
        // been reset. Thermal energy is the only value that is dependent on initial total energy.
        const mass = model.firstSpring.massAttachedProperty.get();
        if ( mass ) {
          mass.initialTotalEnergyProperty.set( mass.kineticEnergyProperty.get() +
                                               mass.gravitationalPotentialEnergyProperty.get() +
                                               mass.elasticPotentialEnergyProperty.get() );
        }
      },
      scale: 0.7
    } );
```

pendulum-lab/EnergyGraphAccordionBox

```ts
    const clearThermalButton = new MoveToTrashButton( {
      arrowColor: PendulumLabConstants.THERMAL_ENERGY_COLOR,
      listener: thermalEnergyProperty.reset.bind( thermalEnergyProperty ),
      scale: 0.72
    } );
```

Usage in Energy Skate Park
<img width="360" alt="image" src="https://user-images.githubusercontent.com/679486/176185929-2ff30d72-936d-4ad6-8959-ad6ab26cb74d.png">

Usage in Pendulum Lab:
<img width="167" alt="image" src="https://user-images.githubusercontent.com/679486/176186347-b427c4da-8ac0-441a-8b1d-44d106a612c4.png">

I observed that the button is created but never displayed in Masses & Springs.  I'll create a side issue for that.  But here's a snapshot of what it used to look like:

<img width="300" alt="image" src="https://user-images.githubusercontent.com/5863899/44506731-f3154d00-a664-11e8-84a7-132a16a07071.png">

If this item is designed to be used as an "inline" button in a legend or chart label, perhaps it has an appropriate set of defaults for its design.  In that case, perhaps the appropriate solution is to factor out the icon so it can be embedded in other styles of buttons where necessary.  Does that sound like a good plan?

**Comment by samreid** [comment]
In the commits, I normalized MoveToTrashButton so now it looks like so:

<img width="417" alt="image" src="https://user-images.githubusercontent.com/679486/183510326-9bdea19e-61b1-4bb0-a01d-300ad3ae647b.png">

It also has a new option that can be used to change the icon size.

In order to support the existing use cases, I factored out a subclass MoveToTrashLegendButton so that those customizations (flat, muted, etc) are factored out.  This is described in the MoveToTrashLegendButton JSDoc:

```ts
/**
 * When a MoveToTrashLegendButton appears in a legend or as part of a bar chart label, typically it is smaller,
 * flat and with more muted colors.
 *
 * @author Sam Reid (PhET Interactive Simulations)
 */
```

I decided not to factor out the icon since that didn't really solve the underlying problems.  Ready for review.
